File: frontend/src/services/careerAi.ts
import { api } from "./api";

export type CareerAIResponse = {
  conversation_id: number;
  answer: string;
  provider: string;
  model: string;
  fallback_used: boolean;
};

export type CareerAIConversation = {
  id: number;
  title: string;
  summary: string;
  pinned: boolean;
  archived: boolean;
  created_at: string;
  updated_at: string;
};

export type CareerAIMessage = {
  id: number;
  conversation_id: number;
  role: "user" | "assistant" | "system";
  content: string;
  provider: string;
  model: string;
  tokens_estimated?: number | null;
  created_at: string;
};

export async function sendCareerAIMessage(message: string, conversationId?: number | null): Promise<CareerAIResponse> {
  const response = await api.post<CareerAIResponse>("/api/career-ai/chat", {
    message,
    conversation_id: conversationId ?? null
  });
  return response.data;
}

export async function listCareerAIConversations(): Promise<CareerAIConversation[]> {
  const response = await api.get<CareerAIConversation[]>("/api/career-ai/conversations");
  return response.data;
}

export async function createCareerAIConversation(title?: string): Promise<CareerAIConversation> {
  const response = await api.post<CareerAIConversation>("/api/career-ai/conversations", { title: title || undefined });
  return response.data;
}

export async function renameCareerAIConversation(conversationId: number, title: string): Promise<CareerAIConversation> {
  const response = await api.patch<CareerAIConversation>(`/api/career-ai/conversations/${conversationId}`, { title });
  return response.data;
}

export async function deleteCareerAIConversation(conversationId: number): Promise<{ ok: boolean }> {
  const response = await api.delete<{ ok: boolean }>(`/api/career-ai/conversations/${conversationId}`);
  return response.data;
}

export async function listCareerAIMessages(conversationId: number): Promise<CareerAIMessage[]> {
  const response = await api.get<CareerAIMessage[]>(`/api/career-ai/conversations/${conversationId}/messages`);
  return response.data;
}

File: backend/models/career_ai.py
from __future__ import annotations

from datetime import datetime
from sqlalchemy import Boolean, DateTime, ForeignKey, Index, Integer, String, Text
from sqlalchemy.orm import Mapped, mapped_column, relationship

from backend.core.database import Base


class CareerAIConversation(Base):
    __tablename__ = "career_ai_conversations"
    __table_args__ = (
        Index("ix_career_ai_conversations_tenant_user_updated", "tenant_id", "user_id", "updated_at"),
    )

    id: Mapped[int] = mapped_column(Integer, primary_key=True)
    tenant_id: Mapped[int] = mapped_column(ForeignKey("tenants.id", ondelete="CASCADE"), index=True, nullable=False)
    user_id: Mapped[int] = mapped_column(ForeignKey("users.id", ondelete="CASCADE"), index=True, nullable=False)
    title: Mapped[str] = mapped_column(String(255), default="Nova conversa", nullable=False)
    summary: Mapped[str] = mapped_column(Text, default="", nullable=False)
    pinned: Mapped[bool] = mapped_column(Boolean, default=False, nullable=False)
    archived: Mapped[bool] = mapped_column(Boolean, default=False, nullable=False)
    created_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow, nullable=False)
    updated_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow, nullable=False)

    messages = relationship("CareerAIMessage", back_populates="conversation", cascade="all, delete-orphan")


class CareerAIMessage(Base):
    __tablename__ = "career_ai_messages"
    __table_args__ = (
        Index("ix_career_ai_messages_conversation_created", "conversation_id", "created_at"),
    )

    id: Mapped[int] = mapped_column(Integer, primary_key=True)
    tenant_id: Mapped[int] = mapped_column(ForeignKey("tenants.id", ondelete="CASCADE"), index=True, nullable=False)
    user_id: Mapped[int] = mapped_column(ForeignKey("users.id", ondelete="CASCADE"), index=True, nullable=False)
    conversation_id: Mapped[int] = mapped_column(ForeignKey("career_ai_conversations.id", ondelete="CASCADE"), index=True, nullable=False)
    role: Mapped[str] = mapped_column(String(24), nullable=False)
    content: Mapped[str] = mapped_column(Text, nullable=False)
    provider: Mapped[str] = mapped_column(String(80), default="", nullable=False)
    model: Mapped[str] = mapped_column(String(120), default="", nullable=False)
    tokens_estimated: Mapped[int | None] = mapped_column(Integer, nullable=True)
    created_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow, nullable=False)

    conversation = relationship("CareerAIConversation", back_populates="messages")
